test/integration/pkg/client/msp/enrollment_test.go

```diff
@@ -304,8 +305,124 @@ func TestEnrollWithProfile(t *testing.T) {
 	if err != nil {
 		t.Fatalf("Enroll failed: %s", err)
 	}
+
 }
 
+func TestEnrollWithCSR(t *testing.T) {
+	// Instantiate the SDK
+	sdk, err := fabsdk.New(integration.ConfigBackend)
+	if err != nil {
+		t.Fatalf("SDK init failed: %s", err)
+	}
+	defer sdk.Close()
+
+	// Delete all private keys from the crypto suite store
+	// and users from the user store at the end
+	integration.CleanupUserData(t, sdk)
+	defer integration.CleanupUserData(t, sdk)
+
+	ctxProvider := sdk.Context()
+
+	// Get the Client.
+	// Without WithOrg option, uses default client organization.
+	mspClient, err := msp.New(ctxProvider)
+	if err != nil {
+		t.Fatalf("failed to create CA client: %s", err)
+	}
+
+	// As this integration test spawns a fresh CA instance,
+	// we have to enroll the CA registrar first. Otherwise,
+	// CA operations that require the registrar's identity
+	// will be rejected by the CA.
+	registrarEnrollID, registrarEnrollSecret := getRegistrarEnrollmentCredentials(t, ctxProvider)
+
+	err = mspClient.Enroll(registrarEnrollID, msp.WithSecret(registrarEnrollSecret))
+	if err != nil {
+		t.Fatalf("Enroll failed: %s", err)
+	}
+
+	// Generate a random user name
+	username := integration.GenerateRandomID()
+
+	// Register the new user
+	enrollmentSecret, err := mspClient.Register(&msp.RegistrationRequest{
+		Name: username,
+		Type: IdentityTypeUser,
+		// Affiliation is mandatory. "org1" and "org2" are hardcoded as CA defaults
+		// See https://github.com/hyperledger/fabric-ca/blob/release/cmd/fabric-ca-server/config.go
+		Affiliation: "org2",
+	})
+	if err != nil {
+		t.Fatalf("Registration failed: %s", err)
+	}
+
+	extraHosts := []string{"localhost"}
```

Comment:
Need to test with more values in extraHosts, it should add all of them to the cert....

Reply:
Can this comparison be O(n²) in test or should it be optimized with a map-based search? 
I mean can I compare all requested hosts with all received hosts or convert requested hosts to a map first to compare? Does it makes difference? Just to write it right from start haha

Reply:
Haha, you can implement comparison any way you want, I am just looking for having at least two values there, just to have a little more confidence that the mapping works correctly through the layers all the way to the CA itself.

Reply:
Ok. Understood.